OpenSim/Common/Component.cpp

```diff
@@ -1526,11 +1529,39 @@ void Component::dumpConnections() const {
 
 
 void Component::initComponentTreeTraversal(const Component &root) const {
-    // Going down the tree, node is followed by all its
-    // children in order, last child's successor is the parent's successor.
+    // Going down the tree, this node is followed by all its children.
+    // The last child's successor (next) is the parent's successor.
+
+    const size_t nmsc = _memberSubcomponents.size();
+    const size_t npsc = _propertySubcomponents.size();
+    const size_t nasc = _adoptedSubcomponents.size();
+
+    // If this isn't the root component and it has no parent, then
+    // this is an orphan component and we likely failed to call 
+    // finalizeFromProperties on the root OR this is a clone that
+    // has not been added to the root (in which case would have a parent).
+    if ((this != &root) && !this->hasParent() ) {
+        OPENSIM_THROW(ComponentIsAnOrphan, getName(), getConcreteClassName());
+    }
+
+    if ((this == &root) && !(nmsc + npsc + nasc)) {
+        ComponentIsRootWithNoSubcomponents ex(__FILE__, __LINE__, __func__,
+            getName(), getConcreteClassName());
+        if (getConcreteClassName() == "Model") {
```

Comment:
I agree other Components can serve as a root, but that is generally not the case. You may create an empty Device and then add to it, in which case the act of probing the initially empty Device you wouldn't want to throw this exception. A Model on the other hand should never be empty (it has Ground, for example) and so is clear in that case that `finalizeFromProperties()` was not called on the model.